In [1]:
import numpy as np
import pandas as pd
import seaborn as sb
import re
import src.cleaning_functions as cf

In [2]:
!pwd

/mnt/c/Users/beasp/IronHack/PROYECTOS/proyecto1_pandas


In [3]:
df_attacks = pd.read_csv("data/attacks.csv",encoding = "ISO-8859-1") #To load the data csv
#mi dataFrame es df_attacks

Hipóteis:
- Se dan más ataques a hombres que practican surf que en hombres que practican natación
- Se dan más ataques en junio que en febrero en California
- El tiburón White shark es el que provoca más ataques en USA

1.Ahora voy a eliminar las columnas que no me van a dar ninguna información para comprobar mis hipótesis, en principio las que voy a usar SEGURO son:
- Sex
- Activity
- Date
- Area
- Species
- Country

In [4]:
df_attacks.head(5)

Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   ...  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F  ...   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F  ...   
2                    Habush, Oahu      Surfing      John Denges    M  ...   
3              Arrawarra Headland      Surfing             male    M  ...   
4                        La Ticla  Free diving   Gustavo Ramos     M  ...   

          Species           Investigator or Source                       pdf  \
0      White shark                R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1              NaN  K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2              NaN  K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3        2 m shark                  B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4  Tiger shark, 3m                       A .Kipper      2018.06.04-Ramos.pdf   

                                        href formula  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...   
1  http://sharkattackfile.net/spreadsheets/pdf_di...   
2  http://sharkattackfile.net/spreadsheets/pdf_di...   
3  http://sharkattackfile.net/spreadsheets/pdf_di...   
4  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                href Case Number.1  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.18   
2  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.09   
3  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.08   
4  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.04   

  Case Number.2 original order Unnamed: 22 Unnamed: 23  
0    2018.06.25         6303.0         NaN         NaN  
1    2018.06.18         6302.0         NaN         NaN  
2    2018.06.09         6301.0         NaN         NaN  
3    2018.06.08         6300.0         NaN         NaN  
4    2018.06.04         6299.0         NaN         NaN  

[5 rows x 24 columns]

In [5]:
borrar= df_attacks.drop(["pdf","href formula","href","Case Number.1","Case Number.2","original order","Unnamed: 22","Unnamed: 23"], axis=1)
#meto en variable "borrar" para borrar las columnas que sé que no me van a hacer falta para mis hipótesis

In [6]:
borrar.head()

Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F   57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F   11   
2                    Habush, Oahu      Surfing      John Denges    M   48   
3              Arrawarra Headland      Surfing             male    M  NaN   
4                        La Ticla  Free diving   Gustavo Ramos     M  NaN   

                                              Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and pad...           N   
1                         Minor injury to left thigh           N   
2       Injury to left lower leg from surfboard skeg           N   
3                          Minor injury to lower leg           N   
4  Lacerations to leg & hand shark PROVOKED INCIDENT           N   

            Time         Species           Investigator or Source  
0          18h00      White shark                R. Collier, GSAF  
1  14h00  -15h00              NaN  K.McMurray, TrackingSharks.com  
2          07h45              NaN  K.McMurray, TrackingSharks.com  
3            NaN        2 m shark                  B. Myatt, GSAF  
4            NaN  Tiger shark, 3m                       A .Kipper

In [7]:
df_attacks.shape

(25723, 24)

2.Voy a eliminar las filas duplicadas y filas con valores vacíos

In [8]:
borrar_duplicados= borrar.drop_duplicates() #me creo una variable para poder seguir trabajando con este dato, como por ejemplo para saber su shape a continuación
borrar_duplicados.head()

Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F   57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F   11   
2                    Habush, Oahu      Surfing      John Denges    M   48   
3              Arrawarra Headland      Surfing             male    M  NaN   
4                        La Ticla  Free diving   Gustavo Ramos     M  NaN   

                                              Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and pad...           N   
1                         Minor injury to left thigh           N   
2       Injury to left lower leg from surfboard skeg           N   
3                          Minor injury to lower leg           N   
4  Lacerations to leg & hand shark PROVOKED INCIDENT           N   

            Time         Species           Investigator or Source  
0          18h00      White shark                R. Collier, GSAF  
1  14h00  -15h00              NaN  K.McMurray, TrackingSharks.com  
2          07h45              NaN  K.McMurray, TrackingSharks.com  
3            NaN        2 m shark                  B. Myatt, GSAF  
4            NaN  Tiger shark, 3m                       A .Kipper

In [33]:
borrar_duplicados.dropna(how="all",inplace=True) #solo había una fila coon todos los valores vacíos
borrar_duplicados.shape

(6304, 16)

Hemos pasado de tener (25723 filas, 24) a tener (6312 filas, 24)

3.Ahora voy a reemplazar los nan por Unknown/Invalid para cada una de las columnas

In [35]:
borrar_duplicados["Case Number"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Date"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Year"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Type"].fillna(("Invalid"),inplace=True)
borrar_duplicados["Country"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Area"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Location"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Activity"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Name"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Sex "].fillna(("Unknown"),inplace=True)
#borrar_duplicados["Age"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Injury"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Fatal (Y/N)"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Time"].fillna(("Unknown"),inplace=True)
borrar_duplicados["Species "].fillna(("Unknown"),inplace=True)
borrar_duplicados["Investigator or Source"].fillna(("Unknown"), inplace=True)

In [36]:
borrar_duplicados.head()

Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F   57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F   11   
2                    Habush, Oahu      Surfing      John Denges    M   48   
3              Arrawarra Headland      Surfing             male    M  NaN   
4                        La Ticla  Free diving   Gustavo Ramos     M  NaN   

                                              Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and pad...           N   
1                         Minor injury to left thigh           N   
2       Injury to left lower leg from surfboard skeg           N   
3                          Minor injury to lower leg           N   
4  Lacerations to leg & hand shark PROVOKED INCIDENT           N   

            Time         Species           Investigator or Source  
0          18h00      White shark                R. Collier, GSAF  
1  14h00  -15h00          Unknown  K.McMurray, TrackingSharks.com  
2          07h45          Unknown  K.McMurray, TrackingSharks.com  
3        Unknown        2 m shark                  B. Myatt, GSAF  
4        Unknown  Tiger shark, 3m                       A .Kipper

4.Comienzo con la sustitución de valores dentro de las columnas que me van a dar información para comprobar mis hipótesis.

4.1 COLUMNA SEX: voy a ver los valores únicos de la columna Sex, que al ser valores categóricos realmente solo me van a dar información si aparece uno de los dos géneros, o desconocido.

In [38]:
sex= borrar_duplicados["Sex "].unique()
sex

array(['F', 'M', 'Unknown', 'M ', 'lli', 'N', '.'], dtype=object)

Voy a reemplazar los valores que parece que quieren decir lo mismo para que se llamen de una misma manera, y los que no me dicen nada por "Unknown"

In [39]:
reemplazar1=borrar_duplicados.replace({"M ": "M", "lli":"Unknown",".":"Unknown","N":"Unknown"}) 
reemplazar1.head()

Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F   57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F   11   
2                    Habush, Oahu      Surfing      John Denges    M   48   
3              Arrawarra Headland      Surfing             male    M  NaN   
4                        La Ticla  Free diving   Gustavo Ramos     M  NaN   

                                              Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and pad...     Unknown   
1                         Minor injury to left thigh     Unknown   
2       Injury to left lower leg from surfboard skeg     Unknown   
3                          Minor injury to lower leg     Unknown   
4  Lacerations to leg & hand shark PROVOKED INCIDENT     Unknown   

            Time         Species           Investigator or Source  
0          18h00      White shark                R. Collier, GSAF  
1  14h00  -15h00          Unknown  K.McMurray, TrackingSharks.com  
2          07h45          Unknown  K.McMurray, TrackingSharks.com  
3        Unknown        2 m shark                  B. Myatt, GSAF  
4        Unknown  Tiger shark, 3m                       A .Kipper

In [40]:
reemplazar1["Sex "].unique() #compruebo si se han reemplazado los valores indicados, viendo de nuevo los valores únicos de la columna Sex.


array(['F', 'M', 'Unknown'], dtype=object)

4.2 COLUMNA ACTIVITY: a partir de mi df ""reemplazar1" que ya cuenta con la limpieza de una columna, voy a limpiar la columna Activity.

In [41]:
len(reemplazar1["Activity"].unique())

1531

In [42]:
Actividad=reemplazar1.Activity.unique() #hay que hacerlo lista porque sino no salen todas las actividades.
list(Actividad)


['Paddling',
 'Standing',
 'Surfing',
 'Free diving',
 'Kite surfing',
 'Swimming',
 'Fishing',
 'Walking',
 'Feeding sharks',
 'Boogie boarding',
 'Scuba diving',
 'Paddle-skiing',
 'Body boarding',
 'Windsurfing',
 'Stand-Up Paddleboarding',
 'Wading',
 'Scuba Diving',
 'Kayak fishing for sharks',
 'Cleaning fish',
 'Snorkeling',
 'Spearfishing',
 'Unknown',
 'Diving',
 'Fishing / Wading',
 '2 boats capsized',
 'Night bathing',
 'Surfing ',
 'Kayaking / Fishing',
 'Kayaking',
 'Body surfing',
 'Swimming, poaching abalone',
 'Canoeing',
 'SUP',
 'Skimboarding',
 'Touching a shark',
 'Fishing ',
 'Attempting to lasso a shark',
 'Photo shoot',
 'Paddle boarding',
 'Kakaying',
 'Washing hands',
 'Grabbing shark for a selfie',
 'Kayak fishing',
 'Tagging sharks',
 'Surf skiing ',
 'Surf fishing',
 'Floating',
 'Cage Diving',
 'Surfng',
 'SUP Foil boarding',
 'Lobstering',
 'Fishing in Alabama Deep Fishing Rodeo',
 'Fishing for squid',
 'Fishing for sharks',
 'Floating in tube',
 'Teasing 

In [43]:
lista_actividad=str(list(Actividad))
lista_actividad

'[\'Paddling\', \'Standing\', \'Surfing\', \'Free diving\', \'Kite surfing\', \'Swimming\', \'Fishing\', \'Walking\', \'Feeding sharks\', \'Boogie boarding\', \'Scuba diving\', \'Paddle-skiing\', \'Body boarding\', \'Windsurfing\', \'Stand-Up Paddleboarding\', \'Wading\', \'Scuba Diving\', \'Kayak fishing for sharks\', \'Cleaning fish\', \'Snorkeling\', \'Spearfishing\', \'Unknown\', \'Diving\', \'Fishing / Wading\', \'2 boats capsized\', \'Night bathing\', \'Surfing \', \'Kayaking / Fishing\', \'Kayaking\', \'Body surfing\', \'Swimming, poaching abalone\', \'Canoeing\', \'SUP\', \'Skimboarding\', \'Touching a shark\', \'Fishing \', \'Attempting to lasso a shark\', \'Photo shoot\', \'Paddle boarding\', \'Kakaying\', \'Washing hands\', \'Grabbing shark for a selfie\', \'Kayak fishing\', \'Tagging sharks\', \'Surf skiing \', \'Surf fishing\', \'Floating\', \'Cage Diving\', \'Surfng\', \'SUP Foil boarding\', \'Lobstering\', \'Fishing in Alabama Deep Fishing Rodeo\', \'Fishing for squid\',

In [44]:
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Ff](ish|ISH).*","Fishing",regex=True)
reemplazar1.head()

Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F   57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F   11   
2                    Habush, Oahu      Surfing      John Denges    M   48   
3              Arrawarra Headland      Surfing             male    M  NaN   
4                        La Ticla  Free diving   Gustavo Ramos     M  NaN   

                                              Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and pad...     Unknown   
1                         Minor injury to left thigh     Unknown   
2       Injury to left lower leg from surfboard skeg     Unknown   
3                          Minor injury to lower leg     Unknown   
4  Lacerations to leg & hand shark PROVOKED INCIDENT     Unknown   

            Time         Species           Investigator or Source  
0          18h00      White shark                R. Collier, GSAF  
1  14h00  -15h00          Unknown  K.McMurray, TrackingSharks.com  
2          07h45          Unknown  K.McMurray, TrackingSharks.com  
3        Unknown        2 m shark                  B. Myatt, GSAF  
4        Unknown  Tiger shark, 3m                       A .Kipper

Compruebo si solo hay un valor de "fishing"

In [45]:
Actividad=reemplazar1.Activity.unique() 
list(Actividad)

['Paddling',
 'Standing',
 'Surfing',
 'Free diving',
 'Kite surfing',
 'Swimming',
 'Fishing',
 'Walking',
 'Feeding sharks',
 'Boogie boarding',
 'Scuba diving',
 'Paddle-skiing',
 'Body boarding',
 'Windsurfing',
 'Stand-Up Paddleboarding',
 'Wading',
 'Scuba Diving',
 'Snorkeling',
 'Unknown',
 'Diving',
 '2 boats capsized',
 'Night bathing',
 'Surfing ',
 'Kayaking',
 'Body surfing',
 'Swimming, poaching abalone',
 'Canoeing',
 'SUP',
 'Skimboarding',
 'Touching a shark',
 'Attempting to lasso a shark',
 'Photo shoot',
 'Paddle boarding',
 'Kakaying',
 'Washing hands',
 'Grabbing shark for a selfie',
 'Tagging sharks',
 'Surf skiing ',
 'Floating',
 'Cage Diving',
 'Surfng',
 'SUP Foil boarding',
 'Lobstering',
 'Floating in tube',
 'Teasing a shark',
 'Surf-skiing',
 'Diving for beche-de-mer',
 'Feeding stingrays?',
 'Sea disaster',
 'Swimming / Wading',
 'Scallop diving on hookah',
 'Treading water',
 'Body boarding ',
 'Playing in the water',
 'Body Boarding',
 'Body surfing?',

Efectivamente, solo hay un valor de Fishing.

In [58]:
len(Actividad)

1205

Continúo unificando tipos de actividad, renombrandolos para que no haya tantos valores diferentes que se refieran mismo tipo de actividad.

In [59]:
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Ss](wim|WIM).*","Swimming",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Kk](ite|ITE).*","Kite",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Ww](alking|ALKING).*","Walking",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Bb](oog|OOG).*","Boogie Board",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Bb](ody|ODY).*","Body Boarding",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[wW](ind|IND).*","Wind Surfing",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Bb](oat|OAT).*","Boat",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Ss](hark|HARK).*","Interact with sharks",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Dd](iving|IVING).*","Diving",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Ss](tand|TAND).*","Standing in water",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Pp](addl|ADDL).*","Paddling",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Bb](athing|ATHING).*","Bathing",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Oo](verb|VERB).*","OverBoard",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Bb](athing|ATHING).*","Bathing",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Ff](loat|LOAT).*","Floating",regex=True)
reemplazar1["Activity"]= reemplazar1["Activity"].str.replace(".*[Jj](ump|UMP).*","Jumping",regex=True)

In [48]:
len(Actividad)

1205

In [49]:
reemplazar1["Activity"].unique()

array(['Paddling', 'Standing in water', 'Surfing', 'Diving', 'Kite',
       'Swimming', 'Fishing', 'Walking', 'Interact with sharks',
       'Boogie Board', 'Body Boarding', 'Wind Surfing', 'Wading',
       'Snorkeling', 'Unknown', 'Boat', 'Bathing', 'Surfing ', 'Kayaking',
       'Canoeing', 'SUP', 'Skimboarding', 'Photo shoot', 'Kakaying',
       'Washing hands', 'Surf skiing ', 'Floating', 'Surfng',
       'SUP Foil boarding', 'Lobstering', 'Surf-skiing',
       'Feeding stingrays?', 'Sea disaster', 'Treading water',
       'Playing in the water', 'Playing', 'Sitting in the water',
       'Transatlantic Rowing', 'Rowing', 'Surfing & filming dolphins',
       'Filming', 'Kayaking ', 'Playing with an air mattress',
       'Kneeling in the water', 'Fell into the water', 'Sailing',
       'Filming a documentary', 'Casting a net', 'Playing in the surf',
       'Jumping', 'Kayaking or Wave skiing', 'Jet skiing', 'Wakeboarding',
       'Attempting to fix motor', 'Rescuing', 'Washing his fe

COLUMNA SPECIES: 

In [56]:
reemplazar2= reemplazar1["Species "].unique()

In [57]:
list(reemplazar2)

['White shark',
 'Unknown',
 '2 m shark',
 'Tiger shark, 3m',
 'Tiger shark',
 "Lemon shark, 3'",
 "Bull shark, 6'",
 'Grey reef shark',
 'Invalid incident',
 'Tawny nurse shark, 2m',
 'Shark involvement not confirmed',
 'Questionable',
 '3 m shark',
 'White shark, 3.5 m',
 'White shark, 2.5 m',
 "6' shark",
 'Juvenile bull shark',
 'Bull shark',
 "Tiger shark, 12'",
 'Wobbegong shark',
 '3.5 m shark',
 '1.8 m shark',
 'Blacktip shark',
 'Juvenile white shark,  2.7 to 3.2 m',
 'Bull shark, 2 m',
 'Possibly a wobbegong',
 'Injury believed caused by an eel, not a shark',
 'Galapagos shark?',
 '2m shark',
 'Bull shark, 3 m ',
 'Grey reef shark. 2 m',
 'small shark',
 'Wobbegong shark?',
 'Juvenile nurse shark',
 "Nurse shark. 5'",
 'Tiger shark, female',
 'Some drowned but other may have been killed by blue sharks',
 'White shark, 4.6 m',
 'Cookiecutter shark',
 'Wobbegong shark, 1 m',
 'White shark, 4.5 m',
 'Spinner shark, 4 to 5 feet',
 'Tiger shark, 8 to 10 feet',
 "8' shark",
 'Death

In [52]:
Fecha= borrar_duplicados["Date"].unique()
Fecha

array(['25-Jun-2018', '18-Jun-2018', '09-Jun-2018', ..., '1883-1889',
       '1845-1853', 'Unknown'], dtype=object)

In [53]:
list(Fecha)

['25-Jun-2018',
 '18-Jun-2018',
 '09-Jun-2018',
 '08-Jun-2018',
 '04-Jun-2018',
 '03-Jun-2018',
 '27-May-2018',
 '26-May-2018',
 '24-May-2018',
 '21-May-2018',
 '13-May-2018',
 'May 2018',
 '12-May-2018',
 '09-May-2018',
 'Reported 30-Apr-2018',
 '28-Apr-2018',
 '25-Apr-2018',
 '24-Apr-2018',
 '23-Apr-2018',
 '22-Apr-2018',
 '19-Apr-2018',
 '15-Apr-2018',
 '14-Apr-2018',
 'Reported 10-Apr-2018',
 '09-Apr-2018',
 '05-Apr-2018',
 '03-Apr-2018',
 '31-Mar-2018',
 '14-Mar-2018',
 '9-Mar-2018',
 '24-Feb-2018',
 '23-Feb-2018',
 '18-Feb-2018',
 '15-Feb-2018',
 '14-Feb-2018',
 '11-Feb-2018',
 '03-Feb-2018',
 '01-Feb-2018',
 '28-Jan-2018',
 '21-Jan-2018',
 '14-Jan-2018',
 '13-Jan-2018',
 '12-Jan-2018',
 '05-Jan-2018',
 '31-Dec-2017',
 '30-Dec-2017',
 '21-Dec-2017',
 '09-Dec-2017',
 '30-Nov-2017',
 'Reported 25-Nov-2017',
 '24-Nov-2017',
 '18-Nov-2017',
 'Reported 13-Nov-2017',
 '13-Nov-2017',
 '04-Nov-2017',
 'Reported 31-Oct-2017',
 '28-Oct-2017',
 '26-Oct-2017',
 '23-Oct-2017',
 '22-Oct-2017',

In [54]:
reemplazar1["Date_month"] = reemplazar1["Date"].apply(mes) #estoy importando una función de visual

NameError: name 'mes' is not defined

In [30]:
Area= borrar_duplicados["Area"].unique()
Area

array(['California', 'Georgia', 'Hawaii', 'New South Wales', 'Colima',
       'Pernambuco', 'Florida', 'Queensland', 'South Carolina',
       'Cornwall', 'Westerm Australia', 'Eastern Cape Province',
       'Hua Hin', 'Cocos Island', 'Western Australia',
       'Alifu Alifu Atoll', 'Western Cape Province', 'New Providence',
       'Alagoas', 'Unknown', 'Victoria', 'KwaZulu-Natal',
       'Boi Island, Victoria', 'Galapagos Islands', 'Fernando de Noronha',
       'Sepang', 'Holquin Province', 'Pamplemousses ', 'South Australia',
       'North Island', 'New York', 'Canary Islands', ' Upolu Island',
       'Shizuoka Prefecture', 'Texas', 'Castellón', 'Massachusetts',
       'Red Sea Protectorate', 'New Providence District',
       '40 miles off Grand Bahama Island', 'Ascension Island',
       'New Jersey', 'Majorca', 'Washington', 'Tabasco', 'Anjouan',
       'Ibiza Island', 'Marquesas', 'South Devon', 'New Providence ',
       'Sharjah, ', 'Baja California Sur', 'Saint-Leu', 'South Island

In [31]:
Pais= borrar_duplicados["Country"].unique()
Pais

array(['USA', 'AUSTRALIA', 'MEXICO', 'BRAZIL', 'ENGLAND', 'SOUTH AFRICA',
       'THAILAND', 'COSTA RICA', 'MALDIVES', 'BAHAMAS', 'NEW CALEDONIA',
       'ECUADOR', 'MALAYSIA', 'LIBYA', 'Unknown', 'CUBA', 'MAURITIUS',
       'NEW ZEALAND', 'SPAIN', 'SAMOA', 'SOLOMON ISLANDS', 'JAPAN',
       'EGYPT', 'ST HELENA, British overseas territory', 'COMOROS',
       'REUNION', 'FRENCH POLYNESIA', 'UNITED KINGDOM',
       'UNITED ARAB EMIRATES', 'PHILIPPINES', 'INDONESIA', 'CHINA',
       'COLUMBIA', 'CAPE VERDE', 'Fiji', 'DOMINICAN REPUBLIC',
       'CAYMAN ISLANDS', 'ARUBA', 'MOZAMBIQUE', 'FIJI', 'PUERTO RICO',
       'ITALY', 'ATLANTIC OCEAN', 'GREECE', 'ST. MARTIN', 'FRANCE',
       'PAPUA NEW GUINEA', 'TRINIDAD & TOBAGO', 'KIRIBATI', 'ISRAEL',
       'DIEGO GARCIA', 'TAIWAN', 'JAMAICA', 'PALESTINIAN TERRITORIES',
       'GUAM', 'SEYCHELLES', 'BELIZE', 'NIGERIA', 'TONGA', 'SCOTLAND',
       'CANADA', 'CROATIA', 'SAUDI ARABIA', 'CHILE', 'ANTIGUA', 'KENYA',
       'RUSSIA', 'TURKS & CAICOS', 